pkg/utils/base_conn.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	tcontext "github.com/pingcap/dm/pkg/context"
+	"github.com/pingcap/dm/pkg/log"
+	"github.com/pingcap/dm/pkg/retry"
+	"github.com/pingcap/dm/pkg/terror"
+
+	"go.uber.org/zap"
+)
+
+// BaseConn wraps a connection to DB
+type BaseConn struct {
+	DB *sql.DB
+
+	// for reset
+	DSN string
+}
+
+var _ retry.Strategy = new(BaseConn)
```

Comment:
What is the meaning of this line?
let the compiler to check `BaseConn` implement `Strategy` _interface_?

We never use `Strategy` _interface_ directly, is it necessary to define such an interface?